File: src/main/java/fptd/Params.java
package fptd;

import java.math.BigInteger;

public class Params {

    public static final boolean IS_PRINT_EXE_INFO = true;

    public static final int NUM_SERVER = 7;
    public static final int N = NUM_SERVER;
    public static final int T = (N/2) + 1;
    public static int ITER_TD = 3;

    public static final BigInteger P = new BigInteger("3351951982485649274893506249551461531869841455148098344430890360930441007518386744200468574541725856922507964546621512713438470702986642486608412251521039");

    public static final String IP_King = "127.0.0.1";
    public static final int Port_King = 8874;

    public static final long PRECISE_ROUND = (long)Math.pow(10, 5);

    public static final String FAKE_OFFLINE_DIR = "./offline_data/";

    private static final int exp = 13;
    public static final BigInteger CONSTANT_FOR_LOG = BigInteger.valueOf(10).pow(exp);
    public static final BigInteger FIXED_DIVISOR_FOR_LOG = BigInteger.valueOf(10).pow(exp - 1);

//    public final static String sensingDataFile = "datasets/d_Duck_Identification/answer.csv";
//    public final static String truthFile = "datasets/d_Duck_Identification/truth.csv";
//    public final static boolean isCategoricalData = true;

//        public final static String sensingDataFile = "datasets/s4_Dog_data/answer.csv";
//        public final static String truthFile = "datasets/s4_Dog_data/truth.csv";
//        public final static boolean isCategoricalData = true;

        public final static String sensingDataFile = "datasets/weather/answer.csv";
        public final static String truthFile = "datasets/weather/truth.csv";
        public final static boolean isCategoricalData = false;
}

File: python_src/fptd/params.py
"""
全局参数配置
"""

class Params:
    # 服务器数量
    NUM_SERVER = 7
    
    # Shamir秘密共享参数
    N = 7  # 参与方数量
    T = 4  # 门限值 (N/2 + 1)
    
    # 真值发现迭代次数
    ITER_TD = 3
    
    # 有限域素数 (512位)
    P = int(
        "686479766013060971498190079908139321726943530014330540939446345918554318339765605212255964066145455497729631139148085803712198799971664381257402855770649"
        "18191753095488370057523717020191526968292064814554920952659462848194611371602116714684652611658559090067375008469446209401503990128824814548817698963319"
        "28686613"
    )
    
    # 定点数精度
    PRECISE_ROUND = 100000
    
    # 数据文件路径
    SENSING_DATA_FILE = "datasets/weather/answer.csv"
    TRUTH_FILE = "datasets/weather/truth.csv"
    IS_CATEGORICAL_DATA = False
    
    # 是否打印执行信息
    IS_PRINT_EXE_INFO = True
    
    # 离线数据目录
    OFFLINE_DATA_DIR = "./offline_data"
    
    # 网络配置
    BASE_PORT = 8000
    HOST = "localhost"
    
    # 除法门参数
    DIV_L = 64      # 除数位长
    DIV_E = 90      # 被除数位长
    DIV_SIGMA = 64  # 安全参数
